src/pretty.jl

```diff
@@ -347,11 +347,55 @@ end
 p_stringh(style::S, cst::CSTParser.EXPR, s::State) where {S<:AbstractStyle} =
     p_stringh(DefaultStyle(style), cst, s)
 
+
+format_docstrings!(something_else) = nothing
+
+function format_docstrings!(document::MD)
+    for clause in document.content
+        format_docstrings!(clause)
+    end
+    nothing
+end
+
+function format_docstrings!(doctest::Code)
+    language = doctest.language
+    code = doctest.code
+    doctest.code = if startswith(language, "@example") ||
+        startswith(language, "@repl") ||
+        startswith(language, "@eval")
+        format_text(code)
+    elseif startswith(language, "jldoctest")
+        if occursin(r"^julia> "m, code)
+            join(
+                (
+                    string("julia> ", join(
+                        (
+                            line for line in split(format_text(input), '\n')
+                        ),
+                        "\n       "
+                    )) for (input, output) in repl_splitter(code)
+                ),
+                "\n\n"
+            )
+        else
+            an_input, output = split(code, "# output\n", limit = 2)
+            string(format_text(String(an_input)),  "# output\n", output)
+        end
+    else
+        code
+    end
+    nothing
+end
+
 # MacroCall
 function p_macrocall(ds::DefaultStyle, cst::CSTParser.EXPR, s::State)
     style = getstyle(ds)
     t = FST(cst, nspaces(s))
     if cst[1].typ === CSTParser.GlobalRefDoc
+        # format docstrings
+        docstring = Markdown.parse(cst[2].val)
+        format_docstrings!(docstring)
```

Comment:
I'll need to check this out but I think this should be called on `.val` returned from p_literal and p_stringh below.

Reply:
I think doing this right would require passing the parser style and state into the inner call to `format_text`. I tried doing that but got lost

Reply:
I checked out the branch and I know what's going on now. So as I mentioned before for strings we don't use `.val` for the CST node. Instead we pull the string directly from the file (this pre-processing is done earlier when the `Document` is initially created).

So I think this should be moved into p_literal and use a keyword arg to label it's from a docstring and extra processing should be done on that string.

https://github.com/domluna/JuliaFormatter.jl/blob/e6152d993b2d723c70246be0e9c8ea050d258da3/src/pretty.jl#L257

this string is wrapped in quotes so it doesn't work right out of the box. We need to extract the inner part and then requote it. 

I can fiddle with this a bit and make a commit.

Reply:
It also looks like just parsing in Markdown and then converting back to a string (without formatting) doesn't guarantee to return the original input, so that's a bit problematic.

Reply:
I'm not sure how to push to this branch directly but https://github.com/domluna/JuliaFormatter.jl/commit/b83b5309917e76945e7fa94e47cf5385446e46a4 expands on this.

Right now it looks like sometimes format_docstrings eats code

```jldoctest
julia> a = 10
10
```

10 isn't printed

It also looks like Markdown.parse does some string escaping/unescaping 
```

Reply:
> 10 isn't printed

Oops sorry that's a bug. I forgot to add the output back in, but you can add an extra `'\n', output` to the end of the string call

> It also looks like just parsing in Markdown and then converting back to a string (without formatting) doesn't guarantee to return the original input

Yeah I thought about that. Could be a feature because you are reformatting the markdown too